build.gradle

```diff
@@ -34,25 +44,66 @@ artifacts {
     archives sourcesJar
 }
 
-pluginBundle {
-  website = 'https://github.com/danthegoodman/gradle-capsule-plugin'
-  vcsUrl = 'https://github.com/danthegoodman/gradle-capsule-plugin.git'
-  description = project_description
-  tags = ['shade', 'fatjar', 'uberjar', 'onejar', 'capsule']
-
-  plugins {
-    capsulePlugin {
-      id = 'us.kirchmeier.capsule'
-      displayName = 'Gradle Capsule Plugin'
+gradlePlugin {
+    plugins {
+        capsulePlugin {
+            id = 'us.kirchmeier.capsule'
+            implementationClass = 'us.kirchmeier.capsule.CapsuleGradlePlugin'
+        }
     }
-  }
-
-  mavenCoordinates {
-    groupId = 'us.kirchmeier'
-    artifactId = "gradle-capsule-plugin"
-  }
 }
 
 test {
     useTestNG()
 }
+
+def publishedProjects = projects.project
+
+configure(publishedProjects) { subproject ->
+    apply plugin: 'java'
+    apply plugin: 'maven-publish'
+    apply plugin: 'com.jfrog.artifactory'
+
+    task sourceJar(type: Jar, dependsOn: subproject.classes) {
+        classifier = 'sources'
+        from sourceSets.main.allSource
+    }
+
+    publishing {
+        publications {
+            create(subproject.name, MavenPublication) {
+
+                if (subproject.hasProperty("mavenArtifacts")) {
```

Comment:
Heh, this has been copied-and-pasted _directly_ from `corda-gradle-plugins`... :smiling_imp: